app/views/incomeSources/add/BusinessAddedObligations.scala.html

```diff
@@ -0,0 +1,183 @@
+                        </thead>
+
+                        <tbody class="govuk-table__body">
+                            @sources.quarterlyObligationsDatesYearTwo.distinct.sortBy(_.periodKey).map{ x =>
+                                <tr class="govuk-table__row">
+                                    <td class="govuk-table__cell">@(x.inboundCorrespondenceFrom.toLongDate + " " + getMessage("to") + " " + x.inboundCorrespondenceTo.toLongDate)</td>
+                                    <td class="govuk-table__cell">@(x.inboundCorrespondenceDue.toLongDate)</td>
+                                </tr>
+                            }
+                        </tbody>
+                    }
+                </table>
+            </div>
+        }
+
+
+        @if(sources.eopsObligationsDates.nonEmpty){
+            <div class="box-simple">
+                <h3 class="govuk-heading-m"><div class="page_headers"> @getMessage("eops-heading") </div></h3>
+                <p class="govuk-body"> @getMessage("eops-p1") </p>
+
+                <table class="govuk-table">
+                    <thead class="govuk-table__head">
+                    <tr class="govuk-table__row">
+                        <th scope="col" class="govuk-table__header govuk" style="width:55%;">@getMessage("table.heading-1")</th>
+                        <th scope="col" class="govuk-table__header govuk" style="width:30%;">@getMessage("table.heading-2")</th>
+                    </tr>
+                    </thead>
+
+                    <tbody class="govuk-table__body">
+                        @sources.eopsObligationsDates.distinct.sortBy(_.inboundCorrespondenceFrom).map{ x =>
+                            <tr class="govuk-table__row">
+                                <td class="govuk-table__cell">@(x.inboundCorrespondenceFrom.getYear + " " + getMessage("to") + " " + x.inboundCorrespondenceTo.getYear)</td>
+                                <td class="govuk-table__cell">@(x.inboundCorrespondenceDue.toLongDate)</td>
+                            </tr>
+                        }
+                    </tbody>
+                </table>
+            </div>
+        }
+
+
+        <div class="box-simple">
+            <h3 class="govuk-heading-m"><div class="page_headers"> @getMessage("final-declaration-heading") </div></h3>
+            <p class="govuk-body"> @getMessage("final-declaration-text") </p>
+
+            <table class="govuk-table">
+                <thead class="govuk-table__head">
+                <tr class="govuk-table__row">
+                    <th scope="col" class="govuk-table__header govuk" style="width:55%;">@getMessage("table.heading-1")</th>
+                    <th scope="col" class="govuk-table__header govuk" style="width:30%;">@getMessage("table.heading-2")</th>
+                </tr>
+                </thead>
+
+                <tbody class="govuk-table__body">
+                    @sources.finalDeclarationDates.distinct.sortBy(_.inboundCorrespondenceFrom).map{ x =>
```

Comment:
this would be better to do distinct / and sorting before you created view model;